Gems/Atom/RPI/Code/Source/RPI.Public/Image/StreamingImage.cpp

```diff
@@ -290,23 +321,27 @@ namespace AZ
 
         void StreamingImage::QueueExpandToMipChainLevel(size_t mipChainIndex)
         {
-            AZ_Assert(IsStreamable(), "Only streamable StreamingImage's mip chain can be expanded");
             AZ_Assert(mipChainIndex < m_mipChains.size(), "Exceeded number of mip chains.");
 
             // Expand operation - queue streaming of mip chains up to target mip chain index. 
             if (m_state.m_streamingTarget > mipChainIndex)
             {
                 // Start on the next-detailed chain from the streaming target.
                 const size_t mipChainBegin = m_state.m_streamingTarget - 1;
-                const size_t mipChainEnd = mipChainIndex - 1;
+
+                // The streaming target need to set before fetching mip chain asset since it's possible the
+                // asset is ready when fetching which may trigger expanding directly
+                m_state.m_streamingTarget = static_cast<uint16_t>(mipChainIndex);
 
                 // Iterate through to the end chain and queue loading operations on the mip assets.
-                for (size_t i = mipChainBegin; i != mipChainEnd; --i)
+                for (size_t i = mipChainBegin; ; --i)
```

Comment:
Decrementing an unsigned value in a loop always scares me, but maybe this should be a while loop construction instead?

Reply:
I changed the for condition. I wonder how could the while loop remove --i condition?